data/txHandler_test.go

```diff
@@ -530,49 +594,141 @@ func runHandlerBenchmark(maxGroupSize int, b *testing.B) {
 		var txnCounter uint64
 		invalidCounter := 0
-		for wi := range outChan {
-			txnCounter = txnCounter + uint64(len(wi.unverifiedTxGroup))
-			groupCounter++
-			u, _ := binary.Uvarint(wi.unverifiedTxGroup[0].Txn.Note)
-			_, inBad := badTxnGroups[u]
-			if wi.verificationErr == nil {
-				require.False(b, inBad, "No error for invalid signature")
-			} else {
-				invalidCounter++
-				require.True(b, inBad, "Error for good signature")
+		defer func() {
+			if txnCounter > 0 {
+				droppedBacklog, droppedPool := getDropped()
+				b.Logf("Input TPS: %d (delay %f microsec)", tps, float64(rateAdjuster)/float64(time.Microsecond))
+				b.Logf("Verified TPS: %d", uint64(txnCounter)*1000000000/uint64(time.Since(tt)))
+				b.Logf("Time/txn: %d(microsec)", uint64((time.Since(tt)/time.Microsecond))/txnCounter)
+				b.Logf("processed total: [%d groups (%d invalid)] [%d txns]", groupCounter, invalidCounter, txnCounter)
+				b.Logf("dropped: [%d backlog] [%d pool]\n", droppedBacklog, droppedPool)
+				handler.Stop() // cancel the handler ctx
+			}
+		}()
+		stopChan := make(chan interface{})
+		go func() {
+			for {
+				time.Sleep(200 * time.Millisecond)
+				droppedBacklog, droppedPool := getDropped()
+				if int(groupCounter+droppedBacklog+droppedPool) == len(signedTransactionGroups) {
+					// all the benchmark txns processed
+					close(stopChan)
+					return
+				}
+			}
+		}()
+
+		for {
+			select {
+			case wi := <-testResultChan:
+				txnCounter = txnCounter + uint64(len(wi.unverifiedTxGroup))
+				groupCounter++
+				u, _ := binary.Uvarint(wi.unverifiedTxGroup[0].Txn.Note)
+				_, inBad := badTxnGroups[u]
+				if wi.verificationErr == nil {
+					require.False(b, inBad, "No error for invalid signature")
+				} else {
+					invalidCounter++
+					require.True(b, inBad, "Error for good signature")
+				}
+				if groupCounter == uint64(len(signedTransactionGroups)) {
+					// all the benchmark txns processed
+					return
+				}
+			case <-stopChan:
+				return
 			}
-		}
-		if txnCounter > 0 {
-			b.Logf("TPS: %d\n", uint64(txnCounter)*1000000000/uint64(time.Since(tt)))
-			b.Logf("Time/txn: %d(microsec)\n", uint64((time.Since(tt)/time.Microsecond))/txnCounter)
-			b.Logf("processed total: [%d groups (%d invalid)] [%d txns]\n", groupCounter, invalidCounter, txnCounter)
 		}
 	}()
 
 	b.ResetTimer()
 	tt = time.Now()
-	for _, stxngrp := range signedTransactionGroups {
-		blm := txBacklogMsg{rawmsg: nil, unverifiedTxGroup: stxngrp}
-		handler.txVerificationPool.EnqueueBacklog(handler.ctx, handler.asyncVerifySignature, &blm, nil)
+	if useBacklogWorker {
+		for _, tg := range encodedSignedTransactionGroups {
+			handler.processIncomingTxn(tg)
+			time.Sleep(rateAdjuster)
+		}
+	} else {
+		for _, stxngrp := range signedTransactionGroups {
+			blm := txBacklogMsg{rawmsg: nil, unverifiedTxGroup: stxngrp}
+			handler.txVerificationPool.EnqueueBacklog(handler.ctx, handler.asyncVerifySignature, &blm, nil)
+			time.Sleep(rateAdjuster)
```

Comment:
Any impact the execution of the few instructions here will have on the benchmark results has no value in the conclusions.

Reply:
@AlgoAxel  Looking at this again, you do have a valid point here. `txVerificationPool.EnqueueBacklog` is non-blocking, hence my dismissal of its time cost. But, when the pool is saturated, it will block, and your point becomes valid :-) 

Moreover, above, `processIncomingTxn` is more problematic, because it is expensive, and in production code, it will be in parallel, but the benchmark here is doing it sequentially, skewing the results. 

Will address these in a new PR as a follow up to this.